ledger-tool/src/main.rs

```diff
@@ -2965,3 +2996,110 @@ fn main() {
     measure_total_execution_time.stop();
     info!("{}", measure_total_execution_time);
 }
+
+fn record_transactions(
+    recv: crossbeam_channel::Receiver<TransactionStatusMessage>,
+    filename: &Path,
+) {
+    #[derive(Serialize, Deserialize, Default, Clone)]
+    struct Transaction {
+        index: usize,
+        accounts: Vec<String>,
+        instructions: Vec<Instruction>,
+        is_simple_vote_tx: bool,
+        execution_results: Option<TransactionExecutionDetails>,
+    }
+
+    #[derive(Serialize, Deserialize, Clone)]
+    struct Instruction {
+        program_id: String,
+        accounts: Vec<String>,
+        data: String,
+    }
+
+    #[derive(Serialize, Deserialize)]
+    struct SlotDetails {
```

Comment:
I'm wondering if we could/should tie this in with the existing `BankHashDetails` struct. I guess we'd probably want to rename that to something like `SlotDetails` or `BlockDetails`, but the idea would be one common file format + reusing some of the similar stuff.

Reply:
I've re-written it so that the transactions are stored in `BankHashSlotDetails`.  Please let me know what you think.